### PR TITLE
AES-CCM null payload buffers with inSz zero

### PR DESCRIPTION
**Pull request by douzzer**
re ZD#11639

### CODE REVIEW

**Review by dgarske — APPROVED** [review]

File: wolfcrypt/test/test.c [hunk 1]
```diff
@@ -10343,6 +10350,40 @@ WOLFSSL_TEST_SUBROUTINE int aesccm_test(void)
     if (XMEMCMP(pl, pl2, sizeof(pl2)))
         ERROR_OUT(-6520, out);
 
+    /* test empty message as null input and output --
```
Comment (SparkiDev) [inline wolfcrypt/test/test.c:10353]: There are some implementations that don't allow NULL and 0.
Need to exclude these tests in those cased.

File: wolfcrypt/src/aes.c [hunk 2]
```diff
@@ -7981,7 +7985,7 @@ int  wc_AesCcmDecrypt(Aes* aes, byte* out, const byte* in, word32 inSz,
     const word32 wordSz = (word32)sizeof(word32);
 
     /* sanity check on arguments */
-    if (aes == NULL || out == NULL || in == NULL || nonce == NULL
+    if (aes == NULL || (inSz != 0 && (in == NULL || out == NULL)) || nonce == NULL
```
Comment (SparkiDev) [inline wolfcrypt/src/aes.c:7988]: 80 characters?

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfcrypt/test/test.c (hunk 1 above)
Comment (douzzer) [reply wolfcrypt/test/test.c:10353]: Ah, those cases are already excluded, by tolerating `BAD_FUNC_ARG`, but _only_ `BAD_FUNC_ARG`, from `wc_AesCcmEncrypt(enc, NULL /* out */, NULL /* in */, 0 /* inSz */, ...)`.

Iff the `wc_AesCcmEncrypt(enc, NULL, ...)` succeeds, then the `wc_AesCcmDecrypt(enc, NULL, ...)` must succeed.

Note that I also added new empty-message encode-decode cycle that must pass on all targets, that pass as inputs explicit non-NULL pointers to `""`, and non-null output buffers.

Comment (douzzer) [reply wolfcrypt/test/test.c:10353]: I've also just added tests to make sure null pointers with `inSz > 0` always return `BAD_FUNC_ARG`.

File: wolfcrypt/src/aes.c (hunk 2 above)
Comment (douzzer) [reply wolfcrypt/src/aes.c]: whoops, missed a spot.  thanks.  fixed.

File: wolfcrypt/src/aes.c [hunk 3]
```diff
@@ -7879,7 +7883,7 @@ int wc_AesCcmEncrypt(Aes* aes, byte* out, const byte* in, word32 inSz,
...
```
Comment (SparkiDev) [inline wolfcrypt/src/aes.c:7886]: 80 chars

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: wolfcrypt/src/aes.c (hunk 3 above)
Comment (douzzer) [reply wolfcrypt/src/aes.c]: fixed.  searched for others -- plenty but no more new ones.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**